lib/srv/db/aws.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package db
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/utils"
+
+	"github.com/gravitational/trace"
+)
+
+func (s *Server) initRDSRootCert(ctx context.Context, server services.DatabaseServer) error {
+	// If this is not an AWS database, or CA was set explicitly, or it was
+	// already loaded, then nothing to do.
+	if !server.IsAWS() || len(server.GetCA()) != 0 || len(s.rdsCACerts[server.GetRegion()]) != 0 {
+		return nil
+	}
+	// This is a RDS/Aurora instance and CA certificate wasn't explicitly
+	// provided, so try to download it from AWS (or see if it's already
+	// been downloaded).
+	downloadURL := rdsDefaultCAURL
+	if u, ok := rdsCAURLs[server.GetRegion()]; ok {
+		downloadURL = u
+	}
+	bytes, err := s.ensureRDSRootCert(downloadURL)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	s.rdsCACerts[server.GetRegion()] = bytes
```

Comment:
nit: parse the cert to sanity-check that it's valid

Reply:
Good call :+1: